doc/1_applications.dox

```diff
@@ -0,0 +1,160 @@
+GridConnect protocol. This is an ASCII protocol that is used by USB-to-CAN
+interfaces sold by [GridConnect](http://gridconnect.com/).
+
+A normal CAN message has the following format:
+
+:X <_Identifier_> <N> <_Data-0_><_Data-1_>...<_Data-7_>;
+
+These messages are sent within a single network (such as a CAN network or over
+TCP) as binary. But if you're using one of the GridConnect devices, their
+I/O format is text. Here is an actual example:
+
+    :X19545415N0502010202000000;
+
+The ID is 19545415. LCC uses 29-bit CAN identifiers. The data bytes are encoded
+as HEX.
+
+You can also have messages that contains no extra data. For example:
+
+    :X17050415N;
+
+A couple of the sample applications output this format to the console in order
+to display the traffic. But, again, the actual CAN/TCP messages will be
+binary messages.
+
+@section application_list Applications
+
+This section contains a brief description of the sample applications to help you
+get up to speed. They're listed in order in which you might want to tackle them
+rather than alphabetical order.
+
+@subsection hub_application Hub
+
+The hub application will run from the command line and it's a great place to
+start with working on an LCC application. You can send CAN packets to the hub
+over TCP and it will display the packets, and also forward them to any others
+programs listening on the same port. This application knows about CAN packets,
+but nothing about LCC. It will forward all CAN packets, regardless of their
+format.
+
+In addition, you can connect this application to an actual USB-to-CAN interface
+in order to have packets flow between the LCC bus and programs running on your
+desktop computer.
+
+As mentioned above in the @ref application_folders section, you should typically
+build samples from within a platform-specific folder. For example:
+
+    cd ~/openmrn/applications/hub/targets/linux.x86
+    make
+    ./hub -h
+
+This will build the application, and then show you the options for the hub
+application. You can run the hub application without any parameters to start
+capturing and displaying CAN traffic over TCP. The next application,
+async_blink, will send out such traffic.
+
+@subsection async_blink_application Async Blink
+
+This is a more involved application, and is a good place to start with LCC. This
+application can be run from the command line, or it can be hosted on a
+micro controller to act as a very simple node.
+
+This node sends out a "blink" event every second.
+
+The Node ID is defined at the target level rather than the shared level. For
+example, if you look at the linux.x86 target folder, you'll find the following
+in the **NodeId.cxx** file:
+
+*   extern const openlcb::NodeID NODE_ID;
+*   const openlcb::NodeID NODE_ID = 0x050101011410ULL;
+
+* **Note:** Node IDs are required to be unique, so you must obtain your own ID
+before you ship anything using code you built with OpenMRN.
```

Comment:
Maybe had a hyperlink to the OpenLCB organization page(s) where Node ID range requests can be made.

http://registry.openlcb.org/uniqueidranges
http://registry.openlcb.org/requestuniqueidrange